dgl_ptm/dgl_ptm/trade_money.py

```diff
@@ -0,0 +1,73 @@
+import os
+import torch 
+import dgl 
+import dgl.function as fn
+
+def trade_money(agent_graph, method: str):
+    """ Trades money between the different connected agents based on 
+        wealth (k) and savings propensity (lambda). 
+        
+        Two methods are provided for wealth exchange: 
+        1 - Random weighted wealth transfer to all connected neigbours
+        2 - Choose one random neighbour for transfering wealth 
+
+        Args:
+            agent_graph: DGLGraph with agent nodes and edges connecting agents
+            method: String with method choice of 'weighted_transfer' or 'singular_transfer'
+
+        Output:
+            agent_graph.ndata['delta_inc']: Adds node attribute 'delta_inc' with amount of
+                wealth transfered from neighbouring agent nodes to self in this time-step.
+
+    NOTE: Assumes that the following properties are available already:
+        k, lambda, w (for 'weighted_transfer'), zeros, ones, total neighbour count
+    NOTE: All edges are bidirected with uniform weights 'w'
+
+    TODO: Rename variables as per Thijs' updates on notebook
+    """
+    # Calculating disposable wealth
+    agent_graph.ndata['disp_wealth'] = agent_graph.ndata['lambda']*agent_graph.ndata['wealth']
+
+    # Transfer of wealth
+    if method == 'weighted_transfer':
+        _weighted_transfer(agent_graph)
+    elif method == 'singular_transfer':
+        _singular_transfer(agent_graph)
+    else:
+        raise NotImplementedError("Incorrect method received. \
+                         Method needs to be either 'weighted_transfer' or 'singular_transfer'")
+    
+def _weighted_transfer(agent_graph):
+    """
+        Weighted transfer of wealth from each agent (node) to every connected 
+        neighbour agent based on pre-defined edge weights.
+    """
+
+    # Sum all incoming weights
+    agent_graph.ndata['tot_wgt'] = torch.zeros(agent_graph.num_nodes(),1)
```

Comment:
I prefer more verbose variable names; I don't really see the point for shortening things like "total" to "tot" but this is mostly harmless as far as I'm concerned.

It becomes a problem for terms like "per" (per what?), "disp" (disparate? display? distribution P value?) or "trfr".
We don't gain anything from writing shorter lines of code if this decreases human readability.

Things like "wgt" and "msg" that are reused almost everywhere could be exampt from this verbosity.

Reply:
@vanlankveldthijs, the intention was to rename the variables as per your naming convention but at that time, I could not find your notebook online. Is it pushed now? Could you maybe direct me to it? 

I currently have this in the file:
TODO: Rename variables as per Thijs' updates on notebook

to ensure we remember to rename it verbosely. :D

Reply:
I never got round to finishing that notebook at the end of the sprint and now I cannot find it. Oops (:-/).

All I remember is:
* basically spelling out the whole words for most shortened terms (e.g. "tot_wgt" -> "total_weight"),
* adding the _msg prefix to messages (e.g. "g.update_all(fn.u_add_e('tot_wgt','w','tot_wgt'), fn.sum('tot_wgt', 'tot_wgt'))" -> g.update_all(fn.u_add_e('tot_wgt','w','tot_wgt_msg'), fn.sum('tot_wgt_msg', 'tot_wgt')))
* In certain cases adding a bit more to variables of complex concepts (e.g. "weight_ratio_at_source" for the variable on an edge that specifies how big the ratio of the edge's weight is relative to all the edge weights with the same source)